src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/ReleasesController.cs

```diff
@@ -214,60 +214,69 @@ public async Task<ActionResult<List<ReleaseViewModel>>> ListScheduledReleases()
                 .HandleFailuresOrOk();
         }
 
-        [HttpGet("release/{releaseId}/data/{fileId}/import/status")]
+        [HttpGet("release/{releaseId:guid}/data/{fileId:guid}/import/status")]
         public Task<ActionResult<DataImportStatusViewModel>> GetDataUploadStatus(Guid releaseId, Guid fileId)
         {
             return _releaseService
                 .GetDataFileImportStatus(releaseId, fileId)
                 .HandleFailuresOrOk();
         }
 
-        [HttpGet("release/{releaseId}/delete-plan")]
+        [HttpGet("release/{releaseId:guid}/delete-plan")]
         public async Task<ActionResult<DeleteReleasePlan>> GetDeleteReleasePlan(Guid releaseId)
         {
             return await _releaseService
                 .GetDeleteReleasePlan(releaseId)
                 .HandleFailuresOrOk();
         }
 
-        [HttpGet("release/{releaseId}/data/{fileId}/delete-plan")]
+        [HttpGet("release/{releaseId:guid}/data/{fileId:guid}/delete-plan")]
         public async Task<ActionResult<DeleteDataFilePlan>> GetDeleteDataFilePlan(Guid releaseId, Guid fileId)
         {
             return await _releaseService
                 .GetDeleteDataFilePlan(releaseId, fileId)
                 .HandleFailuresOrOk();
         }
 
-        [HttpDelete("release/{releaseId}/data/{fileId}")]
+        [HttpDelete("release/{releaseId:guid}/data/{fileId:guid}")]
         public async Task<ActionResult> DeleteDataFiles(Guid releaseId, Guid fileId)
         {
             return await _releaseService
                 .RemoveDataFiles(releaseId, fileId)
                 .HandleFailuresOrNoContent();
         }
 
-        [HttpPost("release/{releaseId}/data/{fileId}/import/cancel")]
+        [HttpPost("release/{releaseId:guid}/data/{fileId:guid}/import/cancel")]
         public async Task<ActionResult> CancelFileImport(Guid releaseId, Guid fileId)
         {
             return await _dataImportService
                 .CancelImport(releaseId, fileId)
                 .HandleFailuresOr(result => new AcceptedResult());
         }
 
-        [HttpGet("releases/{releaseId}/stage-status")]
+        [HttpGet("releases/{releaseId:guid}/stage-status")]
         public async Task<ActionResult<ReleasePublishingStatusViewModel>> GetReleaseStatusesAsync(Guid releaseId)
         {
             return await _releasePublishingStatusService
                 .GetReleaseStatusAsync(releaseId)
                 .HandleFailuresOrOk();
         }
 
-        [HttpGet("releases/{releaseId}/checklist")]
+        [HttpGet("releases/{releaseId:guid}/checklist")]
         public async Task<ActionResult<ReleaseChecklistViewModel>> GetChecklist(Guid releaseId)
         {
             return await _releaseChecklistService
                 .GetChecklist(releaseId)
                 .HandleFailuresOrOk();
         }
+
+        [HttpPatch("releases/{releaseId:guid}")]
```

Comment:
There's no hint of this being an endpoint specifically for updating just one facet of a Release by looking at the URL alone.  SoI might be confused if I saw 2 ways to update a Release using the URL `releases/{releaseId:guid}` - one that operates via a PUT and this one via a PATCH.

Wondering if it's worth being specific that this is an endpoint dealing specifically with Release Published dates, as per the `UpdateReleasePublished` method name, or if we make the `UpdateReleasePublished` method name and the `ReleasePublishedUpdateRequest` class name more reusable for other PATCHable fileds for the future.

Or merge this into the existing PUT on the `releases/{releaseId:guid}` route.

Or maybe we don't need to any of the above, as I'm not too precious about any particular one of them!  I'd possibly be tempted personally to just change the URL slightly to make it obvious that this is specifically something different from what you'd get with the existing PUT.

Reply:
I was thinking that we could have had multiple patch methods in future each with different request types and that the routing would be clever enough to choose the best candidate based on the body, but a quick experiment reveals this is not the case! 😢 

I think merging it into the existing PUT functionality isn't too nice because it requires specifying the other field values in the update and that's why I decided to use PATCH here to make it simple to update a single field.

I've gone with your suggestion for now to change the URL slightly, adding `/publish` to the end to link it with the `UpdateReleasePublished` method name. I don't think this naming is too RESTful though as `published` is not an entity itself which is being updated.

Doing a quick read up it seems like a [JSON Patch](https://jsonpatch.com/) document is the way to go here so that we can offer the ability to patch any number of fields and have a way of specifying which fields should be updated, so as to not accidently set null values for fields not included in the request.

It seems like a bit overkill for now so will leave the current implementation with just the slight change to the URL.